main/main.go

```diff
@@ -431,20 +430,26 @@ func createApplicationRelayersForSourceChain(
 	// The Listener begins processing messages starting from the minimum height across all the ApplicationRelayers
 	minHeight := uint64(0)
 	for _, relayerID := range database.GetSourceBlockchainRelayerIDs(&sourceBlockchain) {
-		height, err := database.CalculateStartingBlockHeight(
-			logger,
-			db,
-			relayerID,
-			sourceBlockchain.ProcessHistoricalBlocksFromHeight,
-			currentHeight,
-		)
-		if err != nil {
-			logger.Error(
-				"Failed to calculate starting block height",
-				zap.String("relayerID", relayerID.ID.String()),
-				zap.Error(err),
+		// Calculate the starting block height for the relayer only if we're processing historical blocks
+		// Otherwies, the first block we process is the next block after the current height
+		height := currentHeight + 1
+		if cfg.ProcessMissedBlocks {
+			var err error
```

Comment:
can we declare the error with `err := database...`

Reply:
Unfortunately not, because doing so would declare a new variable `height` in the inner scope, shadowing the one in the outer scope.